### PR TITLE
Boards page

### PR DESCRIPTION
**Pull request by Avraham-Sason**

### CODE REVIEW

File: src/app/(main-router)/boards/comps.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    const cameraBoards = CacheStore.cameraBoards();
+    return <div className={`_center gap-3 `}>{cameraBoards.includes(board.type) && <PrintQR board={board} />}</div>;
+};
+
+const PrintQR = ({ board }: BoardProps) => {
+    const { t } = useTranslation();
+    const { onPrintClick, PrintableContent } = usePrintQR(board);
+    return (
+        <>
+            <button className="text-[px]" title={t("print")} onClick={() => onPrintClick()}>
```
Comment (akeylesspro) [inline src/app/(main-router)/boards/comps.tsx:91]: what **text-[px]** without a number mean ?

File: src/app/(main-router)/boards/helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        contentRef: containerRef,
+        bodyClass: "",
+        documentTitle: "QR Code",
+    });
+
+    const generateQRCodeImage = useCallback(async () => {
+        let token: string;
+        if (board.token) {
+            token = board.token;
+            console.log("no update token");
```
Comment (akeylesspro) [inline src/app/(main-router)/boards/helpers.tsx:21]: or remove this comment or change it to "no need to update the token" or something more clear

File: src/app/(main-router)/boards/helpers.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    const onPrintClick = useCallback(async () => {
+        await generateQRCodeImage();
+        handlePrint();
+    }, [generateQRCodeImage, handlePrint]);
+
+    const PrintableContent = () => (
+        <div style={{ display: "none" }}>
+            <div ref={containerRef} className="flex justify-center items-center">
+                {/* eslint-disable-next-line @next/next/no-img-element */}
```
Comment (akeylesspro) [inline src/app/(main-router)/boards/helpers.tsx:51]: need this comment ?

File: src/app/(main-router)/boards/helpers.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }, [board]);
+
+    const onPrintClick = useCallback(async () => {
+        await generateQRCodeImage();
+        handlePrint();
+    }, [generateQRCodeImage, handlePrint]);
+
+    const PrintableContent = () => (
+        <div style={{ display: "none" }}>
+            <div ref={containerRef} className="flex justify-center items-center">
```
Comment (akeylesspro) [inline src/app/(main-router)/boards/helpers.tsx:50]: do you need both **justify-center** and **items-center**

File: src/app/(main-router)/boards/page.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import Boards from "./Boards";
+
+export default function BoardsPage() {
+    return <Boards />;
+}
+
+export const dynamic = "force-dynamic";
```
Comment (akeylesspro) [inline src/app/(main-router)/boards/page.tsx:7]: what is the meaning of this export ? I see few similar usages

File: src/components/config/InitialCache.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+export default function InitialCache() {
+    const setBoards = CacheStore.setBoards();
+    const setCameraBoards = CacheStore.setCameraBoards();
+    useSnapshotBulk(
+        [
+            {
+                collectionName: "boards",
+                onFirstTime: (data) => {
+                    console.log("boards first time :", data);
```
Comment (akeylesspro) [inline src/components/config/InitialCache.tsx:15]: I guess you can removed all those logs here ?

File: public/locales/he/translation.json [hunk 7]
```diff
@@ -28,5 +28,7 @@
     "search": "חיפוש",
     "home": "בית",
     "boards": "מערכות",
-    "home_message": "ברוך הבא {name}"
+    "home_message": "ברוך הבא {name}",
+    "maxRowsLabel1": "מציג :",
```
[the remark this answers is not in the document]
Comment (Avraham-Sason) [reply public/locales/he/translation.json:32]: yes

File: src/app/(main-router)/boards/comps.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
+"use client";
+import { Board, BoardStatus } from "akeyless-types-commons";
+import { useTranslation } from "react-i18next";
+import { timestamp_to_string } from "@/lib/helpers";
```
[the remark this answers is not in the document]
Comment (Avraham-Sason) [reply src/app/(main-router)/boards/comps.tsx:4]: no

File: src/app/(main-router)/boards/comps.tsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    ];
+
+    const formattedData = useMemo(() => {
+        return data.map((val) => {
+            return {
+                ...val,
+                actions: <BoardOptions board={val} />,
+                ui_status: t(BoardStatus[val.status]) || "N/A",
+                ui_uploaded: (
+                    <div className="_ellipsis" title={timestamp_to_string(val.uploaded as Timestamp)} style={{ direction: "ltr" }}>
```
[the remark this answers is not in the document]
Comment (Avraham-Sason) [reply src/app/(main-router)/boards/comps.tsx:41]: fix

File: src/app/(main-router)/boards/comps.tsx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+    return (
+        <div style={{ direction: direction }} className="w-full h-full _center">
+            {formattedData.length ? (
+                <Table
+                    headerStyle={{ backgroundColor: "#0f172a", height: "40px" }}
+                    containerHeaderClassName="h-12 justify-between px-2"
+                    containerClassName="_full"
+                    cellClassName="_ellipsis text-start h-10"
+                    tableContainerClass="flex-1 "
```
[the remark this answers is not in the document]
Comment (Avraham-Sason) [reply src/app/(main-router)/boards/comps.tsx:57]: no

File: src/app/(main-router)/boards/comps.tsx (hunk 1 above)
Comment (Avraham-Sason) [reply src/app/(main-router)/boards/comps.tsx:91]: nothing it is mistake

File: src/app/(main-router)/boards/helpers.tsx (hunk 2 above)
Comment (Avraham-Sason) [reply src/app/(main-router)/boards/helpers.tsx:21]: done

File: src/app/(main-router)/boards/helpers.tsx (hunk 3 above)
Comment (Avraham-Sason) [reply src/app/(main-router)/boards/helpers.tsx:51]: yes

File: src/app/(main-router)/boards/helpers.tsx (hunk 4 above)
Comment (Avraham-Sason) [reply src/app/(main-router)/boards/helpers.tsx:50]: yes is for printing

File: src/app/(main-router)/boards/page.tsx (hunk 5 above)
Comment (Avraham-Sason) [reply src/app/(main-router)/boards/page.tsx:7]: its for next.js its mean that this rout/page is dynamic no static

**Pull request merged**